src/main/drivers/rx/rx_sx1280.c

```diff
@@ -0,0 +1,500 @@
+    sx1280WriteCommandBurst(SX1280_RADIO_SET_TXPARAMS, buf, 2);
+}
+
+void sx1280SetPacketParams(const uint8_t preambleLength, const sx1280_lora_packet_lengths_modes_e headerType, const uint8_t payloadLength, const sx1280_lora_crc_modes_e crc, const sx1280_lora_iq_modes_e invertIQ)
+{
+    uint8_t buf[7];
+
+    buf[0] = preambleLength;
+    buf[1] = headerType;
+    buf[2] = payloadLength;
+    buf[3] = crc;
+    buf[4] = invertIQ;
+    buf[5] = 0x00;
+    buf[6] = 0x00;
+
+    sx1280WriteCommandBurst(SX1280_RADIO_SET_PACKETPARAMS, buf, 7);
+}
+
+void sx1280SetMode(const sx1280_operating_modes_e opMode)
+{
+    uint8_t buf[3];
+
+    switch (opMode)
+    {
+
+    case SX1280_MODE_SLEEP:
+        sx1280WriteCommand(SX1280_RADIO_SET_SLEEP, 0x01);
+        break;
+
+    case SX1280_MODE_CALIBRATION:
+        break;
+
+    case SX1280_MODE_STDBY_RC:
+        sx1280WriteCommand(SX1280_RADIO_SET_STANDBY, SX1280_STDBY_RC);
+        break;
+        
+    case SX1280_MODE_STDBY_XOSC:
+        sx1280WriteCommand(SX1280_RADIO_SET_STANDBY, SX1280_STDBY_XOSC);
+        break;
+
+    case SX1280_MODE_FS:
+        sx1280WriteCommand(SX1280_RADIO_SET_FS, 0x00);
+        break;
+
+    case SX1280_MODE_RX:
+        buf[0] = 0x00; // periodBase = 1ms, page 71 datasheet, set to FF for cont RX
+        buf[1] = 0xFF;
+        buf[2] = 0xFF;
+        sx1280WriteCommandBurst(SX1280_RADIO_SET_RX, buf, 3);
+        break;
+
+    case SX1280_MODE_TX:
+        //uses timeout Time-out duration = periodBase * periodBaseCount
+        buf[0] = 0x00; // periodBase = 1ms, page 71 datasheet
+        buf[1] = 0xFF; // no timeout set for now
+        buf[2] = 0xFF; // TODO dynamic timeout based on expected onairtime
+        sx1280WriteCommandBurst(SX1280_RADIO_SET_TX, buf, 3);
+        break;
+
+    case SX1280_MODE_CAD:
+        break;
+
+    default:
+        break;
+    }
+}
+
+void sx1280ConfigLoraModParams(const sx1280_lora_bandwidths_e bw, const sx1280_lora_spreading_factors_e sf, const sx1280_lora_coding_rates_e cr)
+{
+    // Care must therefore be taken to ensure that modulation parameters are set using the command
+    // SetModulationParam() only after defining the packet type SetPacketType() to be used
+
+    uint8_t rfparams[3] = {0};
+
+    rfparams[0] = (uint8_t)sf;
+    rfparams[1] = (uint8_t)bw;
+    rfparams[2] = (uint8_t)cr;
+
+    sx1280WriteCommandBurst(SX1280_RADIO_SET_MODULATIONPARAMS, rfparams, 3);
+
+    switch (sf)
+    {
+    case SX1280_LORA_SF5:
+    case SX1280_LORA_SF6:
+        sx1280WriteRegister(0x925, 0x1E); // for SF5 or SF6
+        break;
+    case SX1280_LORA_SF7:
+    case SX1280_LORA_SF8:
+        sx1280WriteRegister(0x925, 0x37); // for SF7 or SF8
+        break;
+    default:
+        sx1280WriteRegister(0x925, 0x32); // for SF9, SF10, SF11, SF12
+    }
+}
+
+void sx1280SetFrequencyHZ(const uint32_t reqFreq)
+{
+    uint8_t buf[3] = {0};
+
+    uint32_t freq = (uint32_t)((double)reqFreq / (double)SX1280_FREQ_STEP);
```

Comment:
This was my fault, and has been cut/pasted from old code. Double not needed here either

Reply:
There are multiple places in the added driver files where double is used. Looks to me like it's not actually necessary anywhere

Reply:
removed doubles from driver code. Seems to be working